### ISSUE TITLE
Scanning Media Library is incredibly slow

### ISSUE THREAD

**Issue by davispuh** [jellyfin/jellyfin#3722]
**Describe the bug**

After update to 10.6.0 library scanning seems to be stuck. It had all scanned previously.
In logs didn't saw anything useful just #3611
And there's thousands of `OnRefreshStart/OnRefreshComplete` lines in log.

**System:**
 - OS: Arch Linux
 - Virtualization: None
 - Clients: Browser
 - Browser: Firefox
 - Jellyfin Version: 10.6.0
 - Playback: N/A
 - Installed Plugins: AudioDB, Bookshelf, Fanart, LDAP-Auth, MusicBrainz, Open Subtitles, Playback Reporting, Reports
 - Reverse Proxy: nginx
 - Base URL: none
 - Networking: NAT
 - Storage: local

**To Reproduce**
1. Have library fully scanned on previous version
2. Update to 10.6.0
3. Run Scan Media Library
4. See library scan stuck

**Expected behavior**
Complete library scanning successfully.

**Logs**
```
[2020-07-28 03:18:08.448 +03:00] [INF] [1] Main: Jellyfin version: "10.6.0"
[2020-07-28 03:18:08.723 +03:00] [INF] [1] Main: Environment Variables: ["[JELLYFIN_LOG_DIR, /var/lib/jellyfin/log]", "[JELLYFIN_DATA_DIRECTORY, /var/lib/jellyfin]", "[JELLYFIN_CACHE_DIRECTORY, /var/cache/jellyfin]", "[JELLYFIN_ADD_OPTS, ]"]
[2020-07-28 03:18:08.728 +03:00] [INF] [1] Main: Arguments: ["/usr/lib/jellyfin/jellyfin.dll", "--datadir", "/var/lib/jellyfin", "--cachedir", "/var/cache/jellyfin", ""]
[2020-07-28 03:18:08.729 +03:00] [INF] [1] Main: Operating system: "Linux"
[2020-07-28 03:18:08.730 +03:00] [INF] [1] Main: Architecture: X64
[2020-07-28 03:18:08.731 +03:00] [INF] [1] Main: 64-Bit Process: True
[2020-07-28 03:18:08.731 +03:00] [INF] [1] Main: User Interactive: True
[2020-07-28 03:18:08.731 +03:00] [INF] [1] Main: Processor count: 24
[2020-07-28 03:18:08.731 +03:00] [INF] [1] Main: Program data path: "/var/lib/jellyfin"
[2020-07-28 03:18:08.731 +03:00] [INF] [1] Main: Web resources path: "/usr/lib/jellyfin/jellyfin-web"
[2020-07-28 03:18:08.731 +03:00] [INF] [1] Main: Application directory: "/usr/lib/jellyfin/"
[2020-07-28 03:18:09.693 +03:00] [INF] [1] Emby.Server.Implementations.AppBase.BaseConfigurationManager: Setting cache path: "/var/cache/jellyfin"
[2020-07-28 03:18:09.836 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loading assemblies
[2020-07-28 03:18:09.981 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "OpenSubtitlesHandler, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Open Subtitles/OpenSubtitlesHandler.dll"
[2020-07-28 03:18:09.982 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "Jellyfin.Plugin.OpenSubtitles, Version=9.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Open Subtitles/Jellyfin.Plugin.OpenSubtitles.dll"
[2020-07-28 03:18:09.983 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "Jellyfin.Plugin.Fanart, Version=5.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Fanart/Jellyfin.Plugin.Fanart.dll"
[2020-07-28 03:18:09.995 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "Jellyfin.Plugin.PlaybackReporting, Version=7.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Playback Reporting/Jellyfin.Plugin.PlaybackReporting.dll"
[2020-07-28 03:18:09.996 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "Jellyfin.Plugin.TMDbBoxSets, Version=6.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/TMDb Box Sets/Jellyfin.Plugin.TMDbBoxSets.dll"
[2020-07-28 03:18:09.997 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "Trakt, Version=10.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Trakt/Trakt.dll"
[2020-07-28 03:18:10.043 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "Jellyfin.Plugin.Reports, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Reports/Jellyfin.Plugin.Reports.dll"
[2020-07-28 03:18:10.044 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "Jellyfin.Plugin.Bookshelf, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Bookshelf/Jellyfin.Plugin.Bookshelf.dll"
[2020-07-28 03:18:10.045 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "LDAP-Auth, Version=9.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/LDAP Authentication/LDAP-Auth.dll"
[2020-07-28 03:18:10.046 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loaded assembly "Novell.Directory.Ldap.NETStandard, Version=3.2.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/LDAP Authentication/Novell.Directory.Ldap.NETStandard.dll"
[2020-07-28 03:18:11.517 +03:00] [WRN] [1] Microsoft.EntityFrameworkCore.Model.Validation: The entity type '"AccessSchedule"' is configured to use schema '"jellyfin"'. SQLite does not support schemas. This configuration will be ignored by the SQLite provider.
[2020-07-28 03:18:11.518 +03:00] [WRN] [1] Microsoft.EntityFrameworkCore.Model.Validation: The entity type '"ActivityLog"' is configured to use schema '"jellyfin"'. SQLite does not support schemas. This configuration will be ignored by the SQLite provider.
[2020-07-28 03:18:11.518 +03:00] [WRN] [1] Microsoft.EntityFrameworkCore.Model.Validation: The entity type '"ImageInfo"' is configured to use schema '"jellyfin"'. SQLite does not support schemas. This configuration will be ignored by the SQLite provider.
[2020-07-28 03:18:11.518 +03:00] [WRN] [1] Microsoft.EntityFrameworkCore.Model.Validation: The entity type '"Permission"' is configured to use schema '"jellyfin"'. SQLite does not support schemas. This configuration will be ignored by the SQLite provider.
[2020-07-28 03:18:11.518 +03:00] [WRN] [1] Microsoft.EntityFrameworkCore.Model.Validation: The entity type '"Preference"' is configured to use schema '"jellyfin"'. SQLite does not support schemas. This configuration will be ignored by the SQLite provider.
[2020-07-28 03:18:11.518 +03:00] [WRN] [1] Microsoft.EntityFrameworkCore.Model.Validation: The entity type '"User"' is configured to use schema '"jellyfin"'. SQLite does not support schemas. This configuration will be ignored by the SQLite provider.
[2020-07-28 03:18:27.619 +03:00] [INF] [1] Main: Kestrel listening on all interfaces
[2020-07-28 03:18:27.909 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Running startup tasks
[2020-07-28 03:18:27.910 +03:00] [INF] [1] Jellyfin.Plugin.PlaybackReporting.TaskCleanDb: TaskCleanDb Loaded
[2020-07-28 03:18:27.912 +03:00] [INF] [1] Jellyfin.Plugin.PlaybackReporting.TaskRunBackup: TaskRunBackup Loaded
[2020-07-28 03:18:27.956 +03:00] [INF] [1] Emby.Server.Implementations.ScheduledTasks.TaskManager: Daily trigger for "Playback Reporting Trim Db" set to fire at 07/29/2020 00:00, which is 20:41:32.0434778 from now.
[2020-07-28 03:18:27.961 +03:00] [INF] [1] Emby.Server.Implementations.ScheduledTasks.TaskManager: Daily trigger for "Extract Chapter Images" set to fire at 07/29/2020 02:00, which is 22:41:32.0383726 from now.
[2020-07-28 03:18:28.955 +03:00] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Found ffmpeg version "4.3.1"
[2020-07-28 03:18:28.997 +03:00] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available "decoders": ["h264", "h264_qsv", "h264_cuvid", "hevc", "hevc_qsv", "hevc_cuvid", "mpeg2video", "mpeg2_qsv", "mpeg2_cuvid", "mpeg4", "mpeg4_cuvid", "msmpeg4", "vc1_qsv", "vc1_cuvid", "vp8_cuvid", "vp8_qsv", "vp9_cuvid", "vp9_qsv", "aac", "ac3", "mp3"]
[2020-07-28 03:18:29.031 +03:00] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available "encoders": ["libx264", "h264_nvenc", "h264_omx", "h264_qsv", "h264_v4l2m2m", "h264_vaapi", "libx265", "hevc_nvenc", "hevc_qsv", "hevc_vaapi", "mpeg4", "msmpeg4", "libvpx", "libvpx-vp9", "aac", "ac3", "libmp3lame", "libopus", "libvorbis", "srt"]
[2020-07-28 03:18:29.064 +03:00] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available hwaccel types: ["vdpau", "cuda", "vaapi", "qsv", "drm"]
[2020-07-28 03:18:29.064 +03:00] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: FFmpeg: Custom: "/usr/bin/ffmpeg"
[2020-07-28 03:18:29.065 +03:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: ServerId: "020f206a01ab45648c07c02afc47349e"
[2020-07-28 03:18:29.486 +03:00] [INF] [15] Emby.Server.Implementations.ApplicationHost: Executed all pre-startup entry points in 0:00:00.4094019
[2020-07-28 03:18:29.486 +03:00] [INF] [15] Emby.Server.Implementations.ApplicationHost: Core startup complete
[2020-07-28 03:18:29.486 +03:00] [INF] [15] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: EventMonitorEntryPoint Running
[2020-07-28 03:18:29.495 +03:00] [INF] [15] Jellyfin.Plugin.PlaybackReporting.Data.ActivityRepository: Sqlite version: 3.31.1
[2020-07-28 03:18:29.496 +03:00] [INF] [15] Jellyfin.Plugin.PlaybackReporting.Data.ActivityRepository: Sqlite compiler options: COMPILER=gcc-5.4.0 20160609,DEFAULT_FOREIGN_KEYS,ENABLE_COLUMN_METADATA,ENABLE_FTS3_PARENTHESIS,ENABLE_FTS4,ENABLE_FTS5,ENABLE_JSON1,ENABLE_RTREE,THREADSAFE=1
[2020-07-28 03:18:29.499 +03:00] [INF] [15] Jellyfin.Plugin.PlaybackReporting.Data.ActivityRepository: Default journal_mode for "/var/lib/jellyfin/data/playback_reporting.db" is "delete"
[2020-07-28 03:18:29.499 +03:00] [INF] [15] Jellyfin.Plugin.PlaybackReporting.Data.ActivityRepository: Initialize PlaybackActivity Repository
[2020-07-28 03:18:29.504 +03:00] [INF] [15] Jellyfin.Plugin.PlaybackReporting.Data.ActivityRepository: PlaybackActivity table schema OK
[2020-07-28 03:18:29.504 +03:00] [INF] [15] Jellyfin.Plugin.PlaybackReporting.Data.ActivityRepository: Expected : "datecreated:datetime|userid:text|itemid:text|itemtype:text|itemname:text|playbackmethod:text|clientname:text|devicename:text|playduration:int"
[2020-07-28 03:18:29.504 +03:00] [INF] [15] Jellyfin.Plugin.PlaybackReporting.Data.ActivityRepository: Received : "datecreated:datetime|userid:text|itemid:text|itemtype:text|itemname:text|playbackmethod:text|clientname:text|devicename:text|playduration:int"
[2020-07-28 03:18:29.742 +03:00] [INF] [15] Emby.Server.Implementations.ApplicationHost: Executed all post-startup entry points in 0:00:00.2562449
[2020-07-28 03:18:29.743 +03:00] [INF] [15] Main: Startup complete 0:00:21.7861418
[2020-07-28 03:18:29.835 +03:00] [INF] [6] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory /mnt/Movies
[2020-07-28 03:18:29.867 +03:00] [INF] [4] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory /mnt/Photo
[2020-07-28 03:18:47.582 +03:00] [INF] [35] Emby.Server.Implementations.HttpServer.HttpListenerHost: WS "::ffff:127.0.0.1" requestseconds
[2020-07-28 03:19:06.186 +03:00] [INF] [35] Emby.Server.Implementations.ScheduledTasks.TaskManager: ExecuteQueuedTasks
[2020-07-28 03:19:27.967 +03:00] [INF] [10] Emby.Server.Implementations.ScheduledTasks.TaskManager: "IntervalTrigger" fired for task: "Scan Media Library"
[2020-07-28 03:19:27.968 +03:00] [INF] [10] Emby.Server.Implementations.ScheduledTasks.TaskManager: Queueing task "RefreshMediaLibraryTask"
[2020-07-28 03:19:27.968 +03:00] [INF] [35] Emby.Server.Implementations.ScheduledTasks.TaskManager: Executing "Scan Media Library"
[2020-07-28 03:19:27.994 +03:00] [INF] [35] Emby.Server.Implementations.IO.LibraryMonitor: Stopping directory watching for path "/mnt/Photo"
[2020-07-28 03:19:27.996 +03:00] [INF] [35] Emby.Server.Implementations.IO.LibraryMonitor: Stopping directory watching for path "/mnt/Movies"
[2020-07-28 03:19:28.002 +03:00] [INF] [35] Emby.Server.Implementations.Library.LibraryManager: Validating media library
[2020-07-28 03:19:28.986 +03:00] [INF] [35] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshStart "13f2c14a8032762b207f77c64f6985b6"
[2020-07-28 03:19:28.987 +03:00] [INF] [35] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshComplete "13f2c14a8032762b207f77c64f6985b6"
[2020-07-28 03:19:28.987 +03:00] [INF] [35] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshStart "9cea7bbc58ea2060c485e628675d9581"
[2020-07-28 03:19:28.989 +03:00] [INF] [35] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshComplete "9cea7bbc58ea2060c485e628675d9581"
[...]
[2020-07-28 03:20:22.622 +03:00] [ERR] [22] Emby.Server.Implementations.Library.LibraryManager: Cannnot get image dimensions for "/mnt/Music/Song/Folder.jpg"
Non-success codec result: Unimplemented
Jellyfin.Drawing.Skia.SkiaCodecException: Exception of type 'Jellyfin.Drawing.Skia.SkiaCodecException' was thrown.
   at Jellyfin.Drawing.Skia.SkiaHelper.EnsureSuccess(SKCodecResult result) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Jellyfin.Drawing.Skia/SkiaHelper.cs:line 21
   at Jellyfin.Drawing.Skia.SkiaEncoder.GetImageSize(String path) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Jellyfin.Drawing.Skia/SkiaEncoder.cs:line 223
   at Emby.Drawing.ImageProcessor.GetImageDimensions(String path) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Drawing/ImageProcessor.cs:line 316
   at Emby.Drawing.ImageProcessor.GetImageDimensions(BaseItem item, ItemImageInfo info) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Drawing/ImageProcessor.cs:line 307
   at Emby.Server.Implementations.Library.LibraryManager.UpdateImages(BaseItem item, Boolean forceUpdate) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Server.Implementations/Library/LibraryManager.cs:line 1909
[2020-07-28 03:20:22.641 +03:00] [ERR] [22] Emby.Server.Implementations.Library.LibraryManager: Cannnot get image dimensions for "/var/lib/jellyfin/metadata/library/a1/a102435cd82cc538d0a4857d7f8992a5/folder.png"
Non-success codec result: InvalidInput
Jellyfin.Drawing.Skia.SkiaCodecException: Exception of type 'Jellyfin.Drawing.Skia.SkiaCodecException' was thrown.
   at Jellyfin.Drawing.Skia.SkiaHelper.EnsureSuccess(SKCodecResult result) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Jellyfin.Drawing.Skia/SkiaHelper.cs:line 21
   at Jellyfin.Drawing.Skia.SkiaEncoder.GetImageSize(String path) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Jellyfin.Drawing.Skia/SkiaEncoder.cs:line 223
   at Emby.Drawing.ImageProcessor.GetImageDimensions(String path) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Drawing/ImageProcessor.cs:line 316
   at Emby.Drawing.ImageProcessor.GetImageDimensions(BaseItem item, ItemImageInfo info) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Drawing/ImageProcessor.cs:line 307
   at Emby.Server.Implementations.Library.LibraryManager.UpdateImages(BaseItem item, Boolean forceUpdate) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Server.Implementations/Library/LibraryManager.cs:line 1909
[2020-07-28 03:20:22.716 +03:00] [ERR] [22] Emby.Server.Implementations.Library.LibraryManager: Cannnot get image dimensions for "/var/lib/jellyfin/metadata/library/59/597a00c86333c1869492833322ee4ae9/folder.png"
Non-success codec result: InvalidInput
Jellyfin.Drawing.Skia.SkiaCodecException: Exception of type 'Jellyfin.Drawing.Skia.SkiaCodecException' was thrown.
   at Jellyfin.Drawing.Skia.SkiaHelper.EnsureSuccess(SKCodecResult result) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Jellyfin.Drawing.Skia/SkiaHelper.cs:line 21
   at Jellyfin.Drawing.Skia.SkiaEncoder.GetImageSize(String path) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Jellyfin.Drawing.Skia/SkiaEncoder.cs:line 223
   at Emby.Drawing.ImageProcessor.GetImageDimensions(String path) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Drawing/ImageProcessor.cs:line 316
   at Emby.Drawing.ImageProcessor.GetImageDimensions(BaseItem item, ItemImageInfo info) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Drawing/ImageProcessor.cs:line 307
   at Emby.Server.Implementations.Library.LibraryManager.UpdateImages(BaseItem item, Boolean forceUpdate) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Server.Implementations/Library/LibraryManager.cs:line 1909
[2020-07-28 03:20:22.778 +03:00] [ERR] [22] Emby.Server.Implementations.Library.LibraryManager: Cannnot get image dimensions for "/mnt/Photo/P123456.ORF"
Non-success codec result: Unimplemented
Jellyfin.Drawing.Skia.SkiaCodecException: Exception of type 'Jellyfin.Drawing.Skia.SkiaCodecException' was thrown.
   at Jellyfin.Drawing.Skia.SkiaHelper.EnsureSuccess(SKCodecResult result) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Jellyfin.Drawing.Skia/SkiaHelper.cs:line 21
   at Jellyfin.Drawing.Skia.SkiaEncoder.GetImageSize(String path) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Jellyfin.Drawing.Skia/SkiaEncoder.cs:line 223
   at Emby.Drawing.ImageProcessor.GetImageDimensions(String path) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Drawing/ImageProcessor.cs:line 316
   at Emby.Drawing.ImageProcessor.GetImageDimensions(BaseItem item, ItemImageInfo info) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Drawing/ImageProcessor.cs:line 307
   at Emby.Server.Implementations.Library.LibraryManager.UpdateImages(BaseItem item, Boolean forceUpdate) in /mnt/AUR/jellyfin/src/jellyfin-10.6.0/Emby.Server.Implementations/Library/LibraryManager.cs:line 1909
[...]
[2020-07-28 03:21:31.335 +03:00] [INF] [35] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshStart "e087774cae717451078ba4ef15e2511a"
[2020-07-28 03:21:31.372 +03:00] [INF] [35] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshComplete "e087774cae717451078ba4ef15e2511a"
[2020-07-28 03:21:31.372 +03:00] [INF] [35] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshStart "76f54896f364a586d9c072977d4a7c8c"
```

**Screenshots**

![attels](https://user-images.githubusercontent.com/651800/88600951-5dac4000-d067-11ea-9eeb-bee2cb744b2c.png)

**Comment by crobibero** [comment]
It's likely not stuck, upgrading to 10.6 causes a blurhash to be generated for every image. After the initial scan on 10.6 future scans should be normal speed.

**Comment by davispuh** [comment]
Looks like you might be right, but something is terribly wrong with performance. How it could be this slow...

In log look at time gap between entries.

```
[2020-07-28 17:36:40.277 +03:00] [INF] [31] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshStart "4c179b8df38cd2c892c8735a8d2e2eb0"
[2020-07-28 19:56:41.274 +03:00] [INF] [31] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshComplete "4c179b8df38cd2c892c8735a8d2e2eb0"
[2020-07-28 19:56:41.275 +03:00] [INF] [31] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshStart "5ba06af9726f0c43f748ff3a86516a51"
[2020-07-28 20:14:56.914 +03:00] [INF] [31] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshComplete "5ba06af9726f0c43f748ff3a86516a51"
[2020-07-28 20:14:56.914 +03:00] [INF] [31] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshStart "81aff09733080966db5f8d5426b4f046"
[2020-07-28 20:56:36.017 +03:00] [INF] [31] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshComplete "81aff09733080966db5f8d5426b4f046"
[2020-07-28 20:56:36.017 +03:00] [INF] [31] MediaBrowser.Providers.Manager.ProviderManager: OnRefreshStart "57ebe333698cdbdb73fca54431a2dc8d"
```

I've been running it on 3.4GHz 24-Core CPU with 32GB RAM for 7 hours and the progress haven't increased even by 0.01%

**Comment by doughnet** [comment]
Same issue I am having.  I have a fresh installation (first impression of Jellyfin isn't that great).

I have only 4 movies in a folder and it has not completed scanning of 1 file.

**Comment by geek-baba** [comment]
I have 15K Movies and 800 TV Shows and its taking 5 to 7 days to scan new library, its running on 20 Cores Xeon server with 388 GB RAM with P5000 Graphics Card. Something needs to be done for future release to address this issue...

```
System:    Distro: Ubuntu 20.04.1 LTS (Focal Fossa) Kernel: 5.4.0-42-generic x86_64 bits: 64 Console: tty 0            
Machine:   Type: Server System: Dell product: Precision 7920 Rack           
CPU:       2x 10-Core: Intel Xeon Silver 4114 type: MT MCP SMP speed: 800 MHz min/max: 800/3000 MHz 
Memory:    RAM: total: 376.39 GiB used: 11.85 GiB (3.1%)
Graphics:  Device-1: NVIDIA GP104GL [Quadro P5000] driver: nvidia v: 450.57 
           Display: server: X.org 1.20.8 driver: modesetting,nvidia tty: 204x61           
Network:   Device-1: Intel Ethernet 10-Gigabit X540-AT2 driver: ixgbe 
           Device-2: Intel Ethernet 10-Gigabit X540-AT2 driver: ixgbe            
Drives:    Local Storage: total: 238.47 GiB used: 260.60 GiB (109.3%) 
           ID-1: /dev/nvme0n1 vendor: Sabrent model: Rocket Q size: 931.51 GiB 
           ID-2: /dev/nvme1n1 vendor: Samsung model: SSD 950 PRO 256GB size: 238.47 GiB
```

**Comment by whalehub** [comment]
> It's likely not stuck, upgrading to 10.6 causes a blurhash to be generated for every image. After the initial scan on 10.6 future scans should be normal speed.

@crobibero This unfortunately doesn't appear to be the case. The first library scan after 10.6 took almost 10 hours to complete, but instead of subsequent scans performing at pre-10.6 speeds, a scan of my movie library, for example, with only one new movie added takes several hours to complete where previously it would finish in under a minute.

**Comment by cvium** [comment]
Is this still an issue in 10.6.2?

**Comment by whalehub** [comment]
@cvium Yes.

**Comment by cvium** [comment]
Any improvements? Scanning is pretty fast for me, but I only have 400 movies and 5000 episodes.

**Comment by whalehub** [comment]
@cvium I started a new library scan yesterday and unfortunately it still takes several hours to complete each time.

**Comment by davispuh** [comment]
Well the main issue I submitted was about library scanning being stuck. It didn't advance not even by `0.1%` in several days. But now I updated to 10.6.2 and my library scanning have been completed so it's not stuck thus I just renamed this issue.

I don't know if my scanning completed because of update to 10.6.2 or because it's been more than a week of library scanning :D

**Comment by whalehub** [comment]
@davispuh Can you start a new library scan and see how long it takes to complete?

**Comment by davispuh** [comment]
Well after first scan has been completed rescanning isn't that bad. Takes 21mins. So I guess first scan is just super slow or after update from 10.5 to 10.6

![attels](https://user-images.githubusercontent.com/651800/89454532-bd39d800-d758-11ea-8fbf-e4fe78a072ea.png)

**Comment by bugfixin** [comment]
I'm curious if this is related to an issue I found - for those with slow scans, do you have a lot of movie files located within the same directory? (i.e. each movie is not in its own directory)?

**Comment by doughnet** [comment]
I put in 5 movies and tried a fresh install. After 12 hours I cancelled it and deleted the software.

**Comment by davispuh** [comment]
> I'm curious if this is related to an issue I found - for those with slow scans, do you have a lot of movie files located within the same directory? (i.e. each movie is not in its own directory)?

Not movies, but I've photo library where couple of hundred photos in each folder.

**Comment by whalehub** [comment]
> I'm curious if this is related to an issue I found - for those with slow scans, do you have a lot of movie files located within the same directory? (i.e. each movie is not in its own directory)?

@bugfixin No, each movie has its own folder.

**Comment by cvium** [comment]
> I put in 5 movies and tried a fresh install. After 12 hours I cancelled it and deleted the software.

@doughnet You did a fresh install and added a library with only 5 movies and it didn't complete the scan within the 12 hours?

**Comment by doughnet** [comment]
> > I put in 5 movies and tried a fresh install. After 12 hours I cancelled it and deleted the software.
> 
> 
> 
> @doughnet You did a fresh install and added a library with only 5 movies and it didn't complete the scan within the 12 hours?
> 
> 

Correct. I've used Plex since it's beta years. But I dislike their limitations on how many sharing users can be done so looking for an alternative.

I did a fresh install at home on a fresh OS (Ubuntu 20.04 LXC Proxmox) and am passing through my graphics card being shared by all my LXC. Did a fresh install and made a local (not even Shared network) folder with 5 movies that were x265 codec. I let it run over night and later in the day I killed the VM since it had not progressed any percentage.

Try adding a new library with 1 movie and see how long it takes.

**Comment by bugfixin** [comment]
@doughnet it sounds like you're running into the same issue found here: https://github.com/jellyfin/jellyfin/issues/3558#issuecomment-669714695 - are you able/willing to build jellyfin yourself for testing the solution?

**Comment by doughnet** [comment]
> @doughnet it sounds like you're running into the same issue found here: [#3558 (comment)](https://github.com/jellyfin/jellyfin/issues/3558#issuecomment-669714695) - are you able/willing to build jellyfin yourself for testing the solution?

doing a native build right now.  i assume that is what you are requesting.

**Comment by bugfixin** [comment]
You'll have to install it, e.g. via apt install ./jellyfin-server_10.6.*.deb (after uninstalling the existing jellyfin-server) - though I'm not certain of the details for doing it on ubuntu specifically.

**Comment by doughnet** [comment]
I don't see a .deb rile after the build. There a specific location it's at after building?

**Comment by bugfixin** [comment]
It might be somewhere like ~/bin/ubuntu.amd64

**Comment by whalehub** [comment]
I've been keeping an eye on the period library scan times for the past few days and noticed that the amount of time it takes has stayed consistent. The scheduled task runs at midnight every day and always takes 10 hours to complete with zero new media added inbetween scans. I've been using Jellyfin since March 2019 on the same hardware and never experienced library scans this long prior to upgrading to 10.6.0+.

**Comment by cvium** [comment]
@whalehub if you are on 10.6.2, can you run the scan with debug logging?

**Comment by whalehub** [comment]
@cvium I'm running Jellyfin inside Docker on Debian 10 using the linuxserver.io image. Jellyfin is served behind Nginx.

**Comment by cvium** [comment]
@whalehub I am wondering if you're encountering the same issue as doughnet albeit in a smaller scale. Lsio uses our .deb file, which I suspect is built using an old SDK, where threads can deadlock.

**Comment by whalehub** [comment]
@cvium Should I give the official Jellyfin image a try?

**Comment by cvium** [comment]
Same issue I think. We need a new release. The fix is available in unstable but unstable is pretty broken right now.

**Comment by whalehub** [comment]
@cvium What about the portable Jellyfin builds? Are they also affected? https://repo.jellyfin.org/releases/server/portable/stable/

**Comment by doughnet** [comment]
added unstable version and it scans within like 10 ms (4 movies)

so something definitely broken in the stable versions.

Unpacking jellyfin-server (20200811.8-unstable) over (10.6.2-1)

**Comment by bugfixin** [comment]
Unstable builds (now) include the deadlock fix mentioned above so I'd definitely recommend that as a step for troubleshooting, but there are still circumstances where the scan may be very slow outside of a complete deadlock like you were experiencing @doughnet

**Comment by ranaldsgift** [comment]
I believe I have the same issue on my end. I actually have library scans disabled (no scheduled task) and only have File Monitoring on. Usually my libraries will show a percentage - seemingly indicating that they are being refreshed, and the log will will with OnRefreshStart/OnRefreshComplete messages. I see messages from when Sonarr/Radarr connects to tell it there is a new file, but these files don't appear until much later when the "scan" seems to be done with that library. I will try the unstable to see if it helps.

**Comment by whalehub** [comment]
@cvium @bugfixin Unfortunately this issue is still present for me in the newly released 10.6.3.

**Comment by Sparh4wk** [comment]
same issue here.. Im switching from plex and library scan is rly slow.. even after initial one.. when sonnar or radarr add new media, jellyfish takes ages to add it to library.. same goes for when I delete some media.. (with real time scan ofc) this is only problem so far.. otherwise it runs very good.

**Comment by whalehub** [comment]
@cvium @bugfixin Is there any debug info I can provide to help narrow down the cause of this issue? It's bit of a chore to use Jellyfin when it takes half a day to scan the libraries even if only a single file was added. I'm using the latest Jellyfin version (10.6.3).

**Comment by cvium** [comment]
@whalehub I don't have time to fix a proper image/build for you, but if you can test your library scanning with 634ee2d1e913277ea8f419aca9d91a8e58426642 (hopefully faster) and then with 2b400c99ef946ef1e52e3f01cb18bc008a369c59 (should be slower), it would be a big help.

**Comment by whalehub** [comment]
@cvium I'm not sure if this information helps, but I set up a second Jellyfin instance today with identical settings, plugins, media files, etc. and was unable to reproduce the issue there. I turned on debug logging on both instances, but all it shows is that the scan of a library with ~100 movies takes exponentially longer on the existing Jellyfin instance (15m) compared to the fresh one (0.7s). It's possible that only Jellyfin instances that were upgraded from a pre-10.6.x version are affected by this issue for whatever reason.

**Comment by cvium** [comment]
The fresh scan seems more like a bug. In my testing, movies with a lot of actors+crew take a really long time to scan. If you scan Bad Boys For Life with the TMDb provider, it will take 5+ seconds due to refreshing/downloading 113 images for the entire crew.

**Comment by whalehub** [comment]
@cvium Yes, I'm talking about scans after the initial scan.

**Comment by whalehub** [comment]
@cvium @bugfixin @Spacetech I'm happy to report that I finally managed to narrow down the root cause of this and some other slowness issues that I've been experiencing since upgrading to 10.6.x. I cloned my Jellyfin instance for testing purposes earlier today, removed all users except for my own and noticed that this immediately fixed all of the following issues:

- The scheduled task "Scan Media Library" takes half a day to finish, up from five minutes on 10.5.x.
- The media overview (what you see right after logging in) takes up to 10 seconds to load, up from 1-2 seconds on 10.5.x.
- The search function takes up to 15 seconds to display results after typing something, up from 1-2 seconds on 10.5.x.

I need to do some more testing to see if the total number of users or the total number of devices is what's causing the issue, but I think it's more likely to be the latter since the number of users hasn't changed significantly over time, while the number of devices has. Just for reference, my instance is about a year old and currently has about 4,000 registered devices because Jellyfin doesn't prune inactive devices.

**Comment by davispuh** [comment]
Another reason to want #3736 and use PostgreSQL :yum:

**Comment by davispuh** [comment]
and #42

**Comment by whalehub** [comment]
Just opened a new issue with details about the bug I'm experiencing: #4364

**Comment by bugfixin** [comment]
@whalehub while I'm sure you've found at least one aspect of the slow library scanning, there are many facets of slow library scanning and I don't think there's going to be one silver bullet for everyone. At least one such issue I've found is with the structure of user libraries (if you have many movies directly in the library folder, as well as many movies in subfolders, jellyfin will scan all subfolders repeatedly to look for movie extras. Multiplied by the number of movies directly in the library folder). 

Not to say what you've found isn't a great start, I just want to make sure it's not hoisted as a complete solution to or reproduction of slow scans as there are many bugs to find down this rabbit hole.

**Comment by spn1025** [comment]
Wow is all i have to say.  I've been pulling my hair out on this one.
10.6.4 Ubuntu repository release on 20.4.1.  

First install I cancelled and uninstalled after 36 hours on the initial scan and only completed 2 of 11 media folders.
Re-installed and set all of the media folders to pull in the media only with NO metadata or image downloads, took about 45 minutes but at least they were in my library now for the initial scan.  Reset all of the options on each folder to pull down the metadata and images and kicked off each folder to "get missing metadata" with "replace existing image".  Ran about 6 hours and only updated about 9 movies.

Saw the post above about removing users.  I had only 1 extra user, an account for my kids.  Deleted the account, and WHAM, the friggin scan is actually running now!

@whalehub Thank you so much for this, i was about ready to switch back to Windows, which i had zero issues with.

**Comment by YodaQ** [comment]
I have the same issue where my scan gets stuck at 90% and sits there for over 24 hours. I initially thought it was because of my system being very low performance but i see everyone of powerful servers having the same problem.

**Comment by cvium** [comment]
This bug has been known for a while now, so it would be helpful if you would include version number, size of library, number of users etc.

If the slow scanning is related to users, then it should be fixed in 10.7 RC1. If any of you would be willing to backup your 10.6 install and try out the RC release, it would be very helpful.

**Comment by n-peugnet** [comment]
I had this problem with 10.6.4 and I could not wait any longer so I decided to install 10.7.0~RC3. The scan's and search's duration are back to normal and everything is snappier. I do not have any other issue with this release candidate by the way, it is perfect.

**Comment by cvium** [comment]
Based on https://github.com/jellyfin/jellyfin/issues/4364#issuecomment-719506753 I think it's safe to close this issue.

Let me know if it's still a problem in the RC.

**Comment by deepzeafish** [comment]
I'm still experiencing this issue with 10.7.7.

I'm running jellyfin on a Raspberry Pi model 3B. It's a fresh setup, and I'm scanning a library of 168 movies.

The system does not show high cpu or network load. There are occasional 'bursts' of cpu load with 30%. Most of the time the system is just idling.

**Comment by meanOtaku** [comment]
> The system does not show high cpu or network load. There are occasional 'bursts' of cpu load with 30%. Most of the time the system is just idling.
same

**Comment by VigibotDev** [comment]
I have a dedicated Jellyfin server on native Debian 11 (no VM and no docker) with i9 9900K (8 cores 16 thread) 64GB of RAM all is very fast but Jellyfin front-end become unusable slow forever when I add large number of TV Show. I must restart the service AND launch a full scan to repair... There is a bug for sure. Software become slow without any ressources usage !

**Comment by exaveal** [comment]
Same problem here. I'm new to Jellyfin and installed version 10.7.7. Library Scan took 2 days for 917 Movies, 7000 Songs, 2125 Episodes of Series.
When adding an album (max. 20 Songs) to the music library, Jellyfin cannot be used for around 30 minutes. The player just shows the loading circle.

That's really frustrating.

**Comment by Shadowghost** [comment]
Are you using local or cloud storage? The latter makes scanning really slow especially if you use NFOs and posters.

**Comment by voarsh2** [comment]
Hmm.
Every time I scan my library after a short while playback is impossible until the scan is finished, or until I cancel/restart Jellyfin....

**Comment by xenago** [comment]
Why has this issue been closed? Scanning renders jellyfin completely unusable until it has completed, even on NVME/optane. How is that acceptable?

**Comment by cvium** [comment]
https://github.com/jellyfin/jellyfin/issues/3722#issuecomment-781984233

**Comment by xenago** [comment]
?

https://github.com/jellyfin/jellyfin/issues/3722#issuecomment-1272740689

https://github.com/jellyfin/jellyfin/issues/3722#issuecomment-1014283432

https://github.com/jellyfin/jellyfin/issues/3722#issuecomment-1101690937

https://github.com/jellyfin/jellyfin/issues/3722#issuecomment-1126884722

https://github.com/jellyfin/jellyfin/issues/3722#issuecomment-1272740689

**Comment by cvium** [comment]
This issue is over 2 years old and is about slow _initial_ scanning. We've made many improvements in this area. Your issue is unrelated.

**Comment by ecwilsonaz** [comment]
It doesn't make sense to close an issue that is still so severe -- my first scan of a 12TB Kodi library is almost at 2 days, and I see countless mentions across forums about this still being an issue.

**Comment by ecwilsonaz** [comment]
Duplicate of https://github.com/jellyfin/jellyfin/issues/2600

**Comment by davispuh** [comment]
When I opened this issue then it was `Scanning Media Library is incredibly slow` and even today, that's still the case so I don't consider this issue to be fixed. Yes some things were fixed and it was improved, yes, it is faster than it was then but it's still not really acceptable. It's probably many different things together causing it being slow rather than single thing that can be fixed alone.

Also it's that I stopped caring and I think that's the case for most people, we have just accepted that's how it is and how it's been for many years. Basically someone needs to put some serious time into looking into why it's so slow and I don't think anyone has really worked on it. And it's not just about first scan either, even next scans seems to be quite slow and it's making whole Web UI slow whenever it's scanning.

**Comment by VigibotDev** [comment]
+1 for "And it's not just about first scan either, even next scans seems to be quite slow and it's making whole Web UI slow whenever it's scanning." the webserver process should not be slowed down by another task, especially since the processor load never exceeds a few percent on a core i9.

**Comment by bill-mcgonigle** [comment]
How can I increase logging to get helpful information about what's going on with the library scan?  10.9.0 daily on Debian Bullseye.

From the existing log, I see ffprobe doing bursts of about 20 files every 3-4 hours.  Each ffprobe takes approximately 0.09 seconds, and the jellyfin processes are using close to 0% CPU and disk I/O is trivial most of the time.  If I strace the processes/threads they're mostly all calling futex() waiting on locks from each other, checking the dotnet CLR pipes in /tmp somewhat often.

Once in a while something touches library.db but it's usually something like an hour between writes.

Seems like something is deadlocked and something is waking it up on occasion, but certainly I don't know this codebase well enough to guess.  At the current rate the music library will be done scanning in about six weeks.

As far as the hardware, if I use the same command to probe the music library, it finishes in under ten minutes:
```
# time find /storage/music -type f -exec /usr/lib/jellyfin-ffmpeg/ffprobe -analyzeduration 200M -i file:"{}" -threads 0 -v warning -print_format json -show_streams -show_format > /dev/null 2>&1  \; 

real    9m0.475s
user    1m13.792s
sys     0m54.103s

```

**Comment by nikolajsheller** [comment]
The media scanning and webserver contention is still at issue, and becomes a showstopper on large libraries.
The webserver becomes unusable when scanning is running for extended periods.

This issue really shouldn't be marked as closed.

**Comment by VigibotDev** [comment]
Slow on scanning / and on dashboard even on a core i9 with a lot unused core !!!!! a process management problem...

**Comment by nikolajsheller** [comment]
I'm digging into the code to see if I can get an idea about where the problem is.

It is not: CPU, IO or network bound.

There seems to be some sort of contention between the web-server thread that generates the HTML for the browser and the media scanner (thread? async?). it is not yet clear to me if it is a shared database or data structure or where contention is arising.

**Comment by VigibotDev** [comment]
When I delete a huge music library, when it go to 99.8% and log activity slow down for the same work,
[INF] Removing item, Type: "MusicArtist", Name: "...", Path: "/....

I get very slow GUI responses,

I tryed:
  <LibraryScanFanoutConcurrency>4</LibraryScanFanoutConcurrency>
  <LibraryMetadataRefreshConcurrency>4</LibraryMetadataRefreshConcurrency>

(because it run on a 8 core / 16 threads)
with a bit better performance but I'm not sure if it's placebo...
there is a strange bottleneck somewhere...

**Comment by GouZi2019** [comment]
Same issue on 10.8.10

"How is that acceptable?"   LOL

I just wonder if the developers themselves use Jellyfin?
What awesome patience..

**Comment by xenago** [comment]
Are there any maintainers available to reopen this issue? It should not be closed as this remains a major problem (and likely the main showstopper for new users).

**Comment by felix920506** [comment]
Hi @xenago , please explain how this one is not a duplicate of #2600

**Comment by xenago** [comment]
@felix920506 Is the other issue intended as the place for all discussion about slow scanning problems? My understanding is that #2600 ("Very slow initial Library scan") is specifically about the first/setup scan being extremely slow, but this issue is about usually slow scan operations on existing libraries - e.g. the last few comments just above mention slowdown when items were deleted from a library

**Comment by felix920506** [comment]
@xenago after confirming with the devs, nothing different happens with the initial scan and subsequent scans. The issue in #2600 is more like scanning is slow when there is a large amount of changed items
